huaweicloud/resource_huaweicloud_as_lifecycle_hook_test.go

```diff
@@ -0,0 +1,262 @@
+		if !ok {
+			return "", fmt.Errorf("Auto Scaling lifecycle hook not found: %s", hook)
+		}
+		if group.Primary.ID == "" || hook.Primary.ID == "" {
+			return "", fmt.Errorf("resource not found: %s/%s", group.Primary.ID, hook.Primary.ID)
+		}
+		return fmt.Sprintf("%s/%s", group.Primary.ID, hook.Primary.ID), nil
+	}
+}
+
+func testASLifecycleHook_base(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_vpc" "test" {
+  name = "vpc-default"
+}
+
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+
+data "huaweicloud_images_image" "test" {
+  name        = "Ubuntu 18.04 server 64bit"
+  most_recent = true
+}
+
+data "huaweicloud_compute_flavors" "test" {
+  availability_zone = data.huaweicloud_availability_zones.test.names[0]
+  performance_type  = "normal"
+  cpu_core_count    = 2
+  memory_size       = 4
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name        = "%s"
+}
+
+resource "huaweicloud_compute_keypair" "test" {
+  name = "%s"
+
+  lifecycle {
+    ignore_changes = [
+      public_key,
+    ]
+  }
+}
+
+resource "huaweicloud_lb_loadbalancer" "test" {
+  name          = "%s"
+  vip_subnet_id = data.huaweicloud_vpc_subnet.test.subnet_id
+}
+
+resource "huaweicloud_lb_listener" "test" {
+  name            = "%s"
+  protocol        = "HTTP"
+  protocol_port   = 8080
+  loadbalancer_id = huaweicloud_lb_loadbalancer.test.id
+}
+
+resource "huaweicloud_lb_pool" "test" {
+  name        = "%s"
+  protocol    = "HTTP"
+  lb_method   = "ROUND_ROBIN"
+  listener_id = huaweicloud_lb_listener.test.id
+}
+
+resource "huaweicloud_as_configuration" "test"{
+  scaling_configuration_name = "%s"
+
+  instance_config {
+    image    = data.huaweicloud_images_image.test.id
+    flavor   = data.huaweicloud_compute_flavors.test.ids[0]
+    key_name = huaweicloud_compute_keypair.test.id
+
+    disk {
+      size        = 40
+      volume_type = "SATA"
+      disk_type   = "SYS"
+    }
+  }
+}
+
+resource "huaweicloud_as_group" "test"{
+  scaling_group_name       = "%s"
+  scaling_configuration_id = huaweicloud_as_configuration.test.id
+  vpc_id                   = data.huaweicloud_vpc.test.id
+
+  networks {
+    id = data.huaweicloud_vpc_subnet.test.id
+  }
+  security_groups {
+    id = huaweicloud_networking_secgroup.test.id
+  }
+  lbaas_listeners {
+    pool_id       = huaweicloud_lb_pool.test.id
+    protocol_port = huaweicloud_lb_listener.test.protocol_port
+  }
+}
+`, rName, rName, rName, rName, rName, rName, rName)
+}
+
+// Please make sure the smn topic is exist in specifies region and project.
+func testASLifecycleHook_basic(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_as_lifecycle_hook" "test" {
+  name                   = "%s"
+  type                   = "ADD"
+  scaling_group_id       = huaweicloud_as_group.test.id
+  default_result         = "ABANDON"
+  notification_topic_urn = "%s"
```

Comment:
we can create a `huaweicloud_smn_topic` resource.

Reply:
The topic_urn of the resource smn_topic is ForceNew, but if we want to cover `notification_topic_urn`, shall we add two smn_topic resources to the base method?